clippy_lints/src/methods/option_map_unwrap_or.rs

```diff
@@ -39,8 +42,8 @@ pub(super) fn lint<'a, 'tcx>(
             }
         }
 
-        // get snippets for args to map() and unwrap_or()
-        let map_snippet = snippet(cx, map_args[1].span, "..");
+        let map_arg_span = map_args[1].span;
+        // get snippet for unwrap_or()
         let unwrap_snippet = snippet(cx, unwrap_args[1].span, "..");
```

Comment:
If you use this in the suggestion, you should `let mut applicability = MaybeIncorrect;` and later use `snippet_with_applicability(cx, span, &mut applicability)` – this will set the applicability to `HasPlaceholders` if the snippet could not be extracted for whatever reason.